pdi-execution-engine/api/src/main/java/org/pentaho/di/engine/api/IOperation.java

```diff
@@ -0,0 +1,31 @@
+package org.pentaho.di.engine.api;
+
+import java.util.List;
+
+/**
+ * IOperation roughly corresponds to a Step in PDI.
+ * An IOperation represents the logical structure of
+ * a step within the trans graph.  Materialization of
+ * an IOperation converts it to an ICallableOperation,
+ * which is associated with the behavior specific to
+ * an IEngine.
+ */
+public interface IOperation {
```

Comment:
I was hoping to hear more of your thoughts around Hops.  It wasn't clear to me whether they added much beyond what was already captured with .getFrom and .getTo in IOperation.

(also, I'm perfectly okay going back to the IStep naming.  I chose IOperation just to keep myself from thinking about things with too much of a PDI mindset)